src/apdu/dispatcher.c

```diff
@@ -44,14 +27,22 @@ int apdu_dispatcher(const command_t *cmd) {
             }
 
             return handler_get_version();
-        case GET_APP_NAME:
+        case GET_ADDRESS:
             if (cmd->p1 != 0 || cmd->p2 != 0) {
                 return io_send_sw(SW_WRONG_P1P2);
             }
 
-            return handler_get_app_name();
-        case GET_PUBLIC_KEY:
-            if (cmd->p1 > 1 || cmd->p2 > 0) {
+            if (!cmd->data) {
```

Comment:
It checks if `cmd->data` is a null pointer (instead of the actual length). Is it correct?